lib/components/viewers/route-details.js

```diff
@@ -42,13 +42,42 @@ class RouteDetails extends Component {
     pattern: PropTypes.shape({ id: PropTypes.string }),
     route: PropTypes.shape({ id: PropTypes.string }),
     setHoveredStop: setHoveredStop.type,
-    setViewedRoute: setViewedRoute.type
+    setViewedRoute: setViewedRoute.type,
+    vehiclePositionRefreshSeconds: PropTypes.number
   }
 
-  componentDidMount = () => {
+  state = {
+    refreshTimer: null
```

Comment:
This should probably be named something more explicit!
```suggestion
    vehiclePositionsRefreshTimer: null
```